llama-index-core/llama_index/core/agent/react/step.py

```diff
@@ -340,14 +340,8 @@ def _infer_stream_chunk_is_final(self, chunk: ChatResponse) -> bool:
             bool: Boolean on whether the chunk is the start of the final response
         """
         latest_content = chunk.message.content
-        if latest_content:
-            if not latest_content.startswith(
-                "Thought"
-            ):  # doesn't follow thought-action format
-                return True
```

Comment:
This checking logic seems weird to me. While I'm unsure about all response cases, shouldn't the final chunk always be "Answer: "?

Reply:
the model could also go off the rails (very common for open-source LLMs to arbitrarily stop following the react format exactly)

Reply:
@leehuwuj this looks good to me - your one-liner fixes the parser for streaming. @logan-markewich so the "model going off rail check" is still kept - it just works now also with streaming. Can you merge that?